pulp_rpm/app/shared_utils.py

```diff
@@ -66,25 +80,34 @@ def is_previous_version(version, target_version):
         bool: True if versions are the same or if the version is older than the target version.
 
     """
+    # if any of the versions are empty, they can't be compared and
+    # a target_version need to be picked
     if version is None or target_version is None:
-        # if any of the versions are empty, they can't be compared and
-        # a target_version need to be picked
         return True
 
+    # Handle equals
+    if version == target_version:
+        return True
+
+    # Handle integers
     if version.isdigit() and target_version.isdigit():
         return int(version) <= int(target_version)
 
+    # Handle 1.2.3
```

Comment:
I think this can be refactored a bit for clarity.  Does this look equivalent to you?  And feel free to rename variables.

```python
    # Handle 1.2.3
    version_components = version.split(".")
    target_version_components = target_version.split(".")

    if len(version_components) == len(target_version_components):
        for (comp_a, comp_b) in zip(version_components, target_version_components):
            if comp_a.isdigit() and comp_b.isdigit():
                # if both strings contain numeric information, convert them to ints and compare
                a = int(comp_a)
                b = int(comp_b)
                if a < b:
                    return True
                else:
                    continue
            else:
                # if one of the versions contains non-numeric information we cannot compare
                return False
        else:
            return a == b

    # len wasn't equal
    return False
```

Reply:
I guess the use of local variables could be problematic.. hmmm.

Reply:
```int(comp_a) == int(comp_b)``` is probably fine

Reply:
ok, let me experiment and see.  

(Note that this isn't anything involving urljoin - I just tripped over the bug in is_previous_version() as a result of adding the unit-test for urlpath_sanitize(). No Good Deed Goes Unpunished :) )

Reply:
lol

Reply:
Repushed, now with more test-combinations!